core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -684,9 +684,13 @@ public Void getResult()
     class AddLowCardinalityDictionaryPageWork
             extends LowCardinalityDictionaryWork<Void>
     {
+        private final int[] combinationIdToPosition;
+        private int nextCombinationIndex;
+
         public AddLowCardinalityDictionaryPageWork(Page page)
         {
             super(page);
+            this.combinationIdToPosition = calculateCombinationIdToPosition(page);
```

Comment:
nit: technically, this should be part of `process` as constructor shouldn't iterate over page positions in `Work`

Reply:
Refactored to assign these on the first call to `process()` in both occurrences.